### PR TITLE
nixos/ipp-usb: add package and module

### PR DESCRIPTION
**Pull request by symphorien**
###### Description of changes

this allow printing and scanning without vendor-specific driver with any AirPrint, WSD, AirScan capable printer, but over USB instead of network.
tested rebased on 22.11 on one (1) printer :)

needs #202874 for scanning

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
will self-merge next week unless someone complains

File: pkgs/os-specific/linux/ipp-usb/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  postInstall = ''
+    # to accomodate the makefile
+    cp $out/bin/ipp-usb .
+    mkdir -p $out
+    make install DESTDIR=$out
+  '';
+
+  meta = {
+    description = "Daemon to use the IPP everywhere protocol with USB printers";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/ipp-usb/default.nix:37]: homepage is missing

File: pkgs/os-specific/linux/ipp-usb/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  version = "0.9.23";
+
+  src = fetchFromGitHub {
+    owner = "openprinting";
+    repo = "ipp-usb";
+    rev = version;
+    sha256 = "sha256-sbPQWKqkTaD3kLNs0noVIzAN9cwDEaULsqO7SMQH2Jo=";
+  };
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/ipp-usb/default.nix:13]:
```suggestion
  postPatch = ''
```

File: pkgs/os-specific/linux/ipp-usb/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    # rebuild with patched paths
+    rm ipp-usb.8
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/ipp-usb/default.nix:15]: Can't we use make -B instead?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/ipp-usb/default.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/os-specific/linux/ipp-usb/default.nix:36]: added

**Pull request merged**